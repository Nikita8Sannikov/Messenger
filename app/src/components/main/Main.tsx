import { createContext } from "react";

import App from "./App";
import MainClient from "../../core/MainClient";
import { useAppDispatch } from "../../hooks/useGlobalState";

export const MainClientContext = createContext<MainClient>(undefined as never);

// Основной компонент, главная задача которого ровно один раз инициализировать ядро бизнес-логики при открытии страницы пользователем
export default function Main() {
    const dispatch = useAppDispatch();

    const mainClient = new MainClient(dispatch);

    return <MainClientContext.Provider value={mainClient}>
        <App />
    </MainClientContext.Provider>
}