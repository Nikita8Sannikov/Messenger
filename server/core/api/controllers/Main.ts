import type { Express, NextFunction, Request, Response } from "express";

import Logger from "@service/logger";
import { ApiRoutes, HTTPStatuses } from "@custom-types/enums";

const logger = Logger("MainController");

// Класс, отвечающий за основное/внешнее API
export default class MainController {
	constructor(private readonly _app: Express) {
		this._init();
	}

	// Слушатели запросов контроллера MainController
	private _init() {
		this._app.get(ApiRoutes.checkHealth, this._checkHealth);
	}

	/**
     * Проверка "здоровья" сервера.
     * Используется в основном для получения информации о том, что сервер запущен и работает без критичных ошибок.
     */
	private async _checkHealth(_: Request, res: Response, next: NextFunction) {
		try {
			logger.debug("_checkHealth success");

			res.status(HTTPStatuses.NoContent).end();
		} catch (error) {
			next(error);
		}
	}
}
