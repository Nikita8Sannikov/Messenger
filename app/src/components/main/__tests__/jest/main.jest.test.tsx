import { render, screen } from "@testing-library/react";

import Main from "@components/main/Main";

jest.mock("@core/MainClient", () => {
    return jest.fn().mockImplementation(() => {
        return {
            init: jest.fn(),
        };
    });
});

jest.mock("@hooks/useGlobalState", () => ({
    useAppDispatch: () => jest.fn()
}));

jest.mock("@components/main/app", () => ({
    __esModule: true,
    default: () => <div data-testid="app">App Component</div>,
}));


describe("Main", () => {
    test("renders correct", () => {
        render(<Main />);
        const appElement = screen.getByTestId("app");
        expect(appElement).toBeInTheDocument();
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<Main></Main>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
});