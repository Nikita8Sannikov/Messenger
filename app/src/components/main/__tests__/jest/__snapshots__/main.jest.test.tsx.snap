// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="app"
  >
    App Component
  </div>
</DocumentFragment>
`;
